src/ValueToggle/ValueToggle.tsx

```diff
@@ -0,0 +1,138 @@
+import styled from 'styled-components'
+
+import { Color, theme } from '../theme'
+import { Box } from '../Box'
+import { Tag } from '../Tag'
+import { Text } from '../Text'
+import React from 'react'
+import { ToggleButton } from './ToggleButton'
+
+export type ValueToggleOption<T> = {
+  label: string
+  value: T
+  tag?: string // optional tag able to show in the toggle if showTag prop is passed
+  isDisabled?: boolean
+}
+
+export type ValueToggleProps<T> = {
+  options: ValueToggleOption<T>[] // you can pass more than 2 options
+  value: T
+  onChange: (value: T) => void
+  showTag?: boolean // pass this prop if you want to show the tag in the toggle
+  toggleColor?: Color
+  backgroundColor?: Color
+  textColor?: Color
+  selectedTextColor?: Color
+  tagColor?: Color
+  tagTextColor?: Color
+  tagBorderColor?: Color
+}
+
+// recommend to use a useState to store the selected value above the component
+export const ValueToggle = <T,>({
```

Comment:
This is probably a nitpicky opinion and can be argued but can we call this component `SegmentedControl`?

I think `SegmentedControl` is a more widely understood name for what this component is. afaik it originates from mobile UI. If you google `SegmentedControl` you will also be able to get images with seemingly similar UI components.

Just to ride the idea of the name home, here's a react native component using the name:
https://github.com/react-native-segmented-control/segmented-control

Reply:
You should be able to quickly refactor by using 'Rename symbol...' refactor command in vscode

Reply:
I med a quick look around on google and it seams you right 👏 the right name for that is `SegmentedControl ` so I renamed it 💯